File: docs-examples/example-java/src/test/java/io/micronaut/jms/docs/configuration/CustomizeBrokerJMSConnectionPoolListener.java
package io.micronaut.jms.docs.configuration;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import jakarta.inject.Singleton;
import org.apache.activemq.ActiveMQConnectionFactory;

import javax.jms.ConnectionFactory;
// end::imports[]

@Requires(property = "spec.name", value = "CustomizeBrokerSpec")
// tag::clazz[]
@Singleton
public class CustomizeBrokerJMSConnectionPoolListener implements BeanCreatedEventListener<ConnectionFactory> {

    @Override
    public ConnectionFactory onCreated(BeanCreatedEvent<ConnectionFactory> event) {

        ConnectionFactory connectionFactory = event.getBean();
        if (connectionFactory instanceof ActiveMQConnectionFactory) {
            ActiveMQConnectionFactory amqcf = (ActiveMQConnectionFactory) connectionFactory;
            amqcf.setUseAsyncSend(true);
        }

        return connectionFactory;
    }
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/jms/docs/configuration/CustomizeBrokerJMSConnectionPoolListener.py
from micronaut.context.annotation import Requires
# tag::imports[]
from jakarta.inject import Singleton
from jakarta.jms import ConnectionFactory
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener
from org.apache.activemq import ActiveMQConnectionFactory
# end::imports[]


@Requires(property="spec.name", value="CustomizeBrokerSpec")
# tag::clazz[]
@Singleton
class CustomizeBrokerJMSConnectionPoolListener(BeanCreatedEventListener[ConnectionFactory]):

    def onCreated(self, event: BeanCreatedEvent[ConnectionFactory]) -> ConnectionFactory:
        connection_factory = event.getBean()
        if isinstance(connection_factory, ActiveMQConnectionFactory):
            connection_factory.setUseAsyncSend(True)
        return connection_factory
# end::clazz[]
